R/decoupleR-benchmark.R

```diff
@@ -0,0 +1,104 @@
+#' Benchmark pipeline built on the statistical method wrapper decouple.
+#'
+#' @inheritParams format_design
+#' @param .form bool whether to format or not
+#' @param .perform bool whether to calculate roc and performance summary
+#' @inheritParams filter_sets
+#' @param .downsample_pr whether to downsample precision recall curve TNs
+#' @param .downsample_roc whether to downsample ROC true negatives
+#' @param .downsample_times downsampling iterattions
+#' @inheritParams readRDS_helper
+#'
+#' @export
+#' @importFrom rlang .data
+#' @importFrom stats reorder setNames
+#' @import tibble tidyr dplyr tidyselect
+#' @seealso See \link{input_tibble} for a description of the params/column
+#' of .design (i.e. input tibble).
+#' @return An S4 object of \link{BenchResult-class}
+run_benchmark <- function(.design,
+                          .form = TRUE,
+                          .perform = TRUE,
+                          .minsize = 10,
+                          .silent = TRUE,
+                          .downsample_pr = FALSE,
+                          .downsample_roc = FALSE,
+                          .downsample_times = 100,
+                          .url_bool = FALSE
+                          ){
+  res <- .design %>%
+    format_design() %>%
+    mutate(activity = pmap(.,
+                           .f=function(set_name, bench_name,
+                                       stats_list, opts_list,
+                                       bexpr_loc, bmeta_loc, source_loc,
+                                       source_col, target_col,
+                                       filter_col, filter_crit,
+                                       .source_bln, .expr_bln, .meta_bln){
+
+      # Check_prereq
```

Comment:
Using or modifying global variables can be dangerous, why is it necessary? Can it be avoided?

For now this should be reported in the details of the function.